File: imports-frontend-entities/src/entities/commodity_risk_result.js
const handler = require('./base/handler')

module.exports = class CommodityRiskResult {
  constructor(obj = {}) {

    this.riskDecision = obj.riskDecision
    this.hmiDecision = obj.hmiDecision
    this.phsiDecision = obj.phsiDecision
    this.phsiClassification = obj.phsiClassification
    this.phsi = obj.phsi
    this.uniqueId = obj.uniqueId
    this.eppoCode = obj.eppoCode
    this.variety = obj.variety
    this.isWoody = obj.isWoody
    this.indoorOutdoor = obj.indoorOutdoor
    this.propagation = obj.propagation
    this.phsiRuleType = obj.phsiRuleType

    return Object.seal(new Proxy(this, handler))
  }
}

File: notification-schema-java/src/main/java/uk/gov/defra/tracesx/notificationschema/representation/CommodityRiskResult.java
package uk.gov.defra.tracesx.notificationschema.representation;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import java.util.UUID;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import uk.gov.defra.tracesx.notificationschema.representation.enumeration.HmiDecision;
import uk.gov.defra.tracesx.notificationschema.representation.enumeration.PhsiClassification;
import uk.gov.defra.tracesx.notificationschema.representation.enumeration.PhsiDecision;
import uk.gov.defra.tracesx.notificationschema.representation.enumeration.PhsiRuleType;
import uk.gov.defra.tracesx.notificationschema.representation.enumeration.RiskDecision;

@Builder
@Data
@JsonInclude(Include.NON_EMPTY)
@NoArgsConstructor(access = AccessLevel.PUBLIC)
@AllArgsConstructor(access = AccessLevel.PUBLIC)
public class CommodityRiskResult {

  private RiskDecision riskDecision;
  private HmiDecision hmiDecision;
  private PhsiDecision phsiDecision;
  private PhsiClassification phsiClassification;
  private Phsi phsi;
  private UUID uniqueId;
  private String eppoCode;
  private String variety;
  private Boolean isWoody;
  private String indoorOutdoor;
  private String propagation;
  private PhsiRuleType phsiRuleType;
}
